tests/pass-dep/libc/libc-fs.rs

```diff
@@ -451,3 +454,20 @@ fn test_ioctl() {
         assert_eq!(libc::ioctl(fd, libc::FIOCLEX), 0);
     }
 }
+
+/// Basic test for `readlink`.
+fn test_readlink() {
+    let mut buf = vec![0; "foo_link.txt".len() + 1];
+    unsafe {
+        assert_eq!(libc::readlink(c"foo.txt".as_ptr(), buf.as_mut_ptr(), buf.len()), -1);
+        assert_eq!(Error::last_os_error().raw_os_error(), Some(libc::ENOENT));
+    }
+}
```

Comment:
Why does this not give a loud error with isolation? That seems like a pretty bad bug?